src/main/java/seedu/address/model/person/insights/MostRecentEventInsight.java

```diff
@@ -0,0 +1,106 @@
+package seedu.address.model.person.insights;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import seedu.address.model.Model;
+import seedu.address.model.event.DateTime;
+import seedu.address.model.event.Event;
+import seedu.address.model.person.Person;
+
+/**
+ * This class encapsulates the insight of the most recent event.
+ */
+public class MostRecentEventInsight extends Insight implements Comparable<MostRecentEventInsight> {
+
+    // data field
+    private final DateTime dateTime;
+    private final boolean hasAtLeastOneEvent;
+
+    private MostRecentEventInsight(boolean hasAtLeastOneEvent) {
+        assert(!hasAtLeastOneEvent);
+        this.hasAtLeastOneEvent = false;
+        this.dateTime = null;
+    }
+
+    private MostRecentEventInsight(DateTime dateTime) {
+        requireNonNull(dateTime);
+        this.dateTime = dateTime;
+        this.hasAtLeastOneEvent = true;
+    }
+
+    private MostRecentEventInsight() {
+        // dummy todo surely there is a better way
+        this.dateTime = null;
+        this.hasAtLeastOneEvent = false;
+    }
+
+    /**
+     * Statically constructs a MostRecentEventInsight object.
+     */
+    public static MostRecentEventInsight of(Person person, Model model) {
+        requireAllNonNull(person, model);
+        MostRecentEventInsight helper = new MostRecentEventInsight(); // dummy todo surely there is a better way
+        return helper.getInsight(person, model);
+    }
+
+    @Override
+    public MostRecentEventInsight getInsight(Person person, Model model) {
+        requireAllNonNull(person, model);
+        List<Event> eventsWithPerson = model.getAddressBook()
+                .getEventList()
+                .stream()
+                .filter(event -> event.hasFriendWithName(person))
+                .collect(Collectors.toList());
+        if (eventsWithPerson.size() > 0) {
+            eventsWithPerson.sort(Event::compareTo);
+            DateTime dateTime = eventsWithPerson.get(eventsWithPerson.size() - 1).getDateTime();
```

Comment:
What if the latest event is upcoming? i.e going to happen and has not happened. Would this still count as the most recent event as it technically has not happened?